### DISCUSSION TITLE
Some notes on the use of `SpaceGroupIrep` and `vasp2trace` for spin orbit coupling and spin polarized calculations.

### DISCUSSION THREAD

**Discussion by hongyi-zhao** [goodluck1982/SpaceGroupIrep#29] [General]
1. In general, we can't do spin orbit coupling and spin polarized calculation simultaneously, as commented [here](https://www.researchgate.net/post/Can_we_do_spin_orbit_coupling_and_spin_polarized_calculation_simultaneously):

```
The term "spin polarized calculations" assumes that the spatial and spin coordinates in one-electron eigenfunctions are
separable, i.e. each one-electron eigenstate is characterized with definite value of one-electron spin projection, being either
 "state-spin up" or "state-spin down". As we include spin-orbital coupling in the Hamiltonian, it is no longer so. The one-
electron eigenstates will be then the linear combinations of "spin up/down states", being neither "spin up", nor "spin down" 
states, so, it would make no sense to tell about "spin polarization", - in general.
```

The following are some relevant comments/discussions:

- [ProcarPy](https://github.com/K4ys4r/ProcarPy#what-procarpy-can-do-):
![image](https://user-images.githubusercontent.com/11155854/160996806-02174f27-2227-4736-9300-5d4ee86f395a.png)
- [PROCAR](https://www.vasp.at/wiki/index.php/PROCAR):
![image](https://user-images.githubusercontent.com/11155854/160996940-f6360ca3-3478-4054-ac19-7f8019fd52e3.png)
- Please also refer to the relevant discussion [here](https://www.vasp.at/forum/viewtopic.php?f=4&p=21707#p21707).
 
2. [vasp2trace](https://github.com/zjwang11/irvsp#vasp2trace) has different versions for spin polarized and non spin polarized calculations:

![image](https://user-images.githubusercontent.com/11155854/153330781-b7f7999f-5a53-477b-b7e9-53e89f95be3b.png)

For the meaning of the parameter, see [here](https://www.vasp.at/wiki/index.php/ISPIN):
```
    ISPIN=1: non spin polarized calculations are performed.
    ISPIN=2: spin polarized calculations (collinear) are performed.
```
3. The `convTraceToBC` function in `SpaceGroupIrep` uses different argument list for SOC and non-SOC cases:

```shell
$ rg -A4 convTraceToBC::
SpaceGroupIrep.wl
237:convTraceToBC::usage="convTraceToBC[sgno,traceData,P,p0,stdR]  converts traceData from non-BC standard to BC "<>
238-   "standard for getBandRep to determinate little group ireps. P, p0, and stdR are respectively "<>
239-   "dataset['transformation_matrix'], dataset['origin_shift'] and dataset['std_rotation_matrix'] "<>
240-   "from spglib acting on the initial cell. For non-SOC case stdR is not needed but for SOC case "<>
241-   "stdR has to be given.";
```
[Here](https://github.com/goodluck1982/SpaceGroupIrep/issues/5#issuecomment-1034444412) are some related comments.

**Comment by goodluck1982** [comment]
>2\. [vasp2trace](https://github.com/zjwang11/irvsp#vasp2trace) has different versions for spin polarized and non spin polarized calculations:

`getBandRep` works for the `trace.txt` (or `trace_up.txt` and `trace_dn.txt`) file generated by all these versions.

> 3\. The `convTraceToBC` function in `SpaceGroupIrep` uses different argument options for SOC and non-SOC cases:

Strictly speaking, the function `convTraceToBC` should be called with the same number of arguments for both SOC and non-SOC. However, when called in noo-SOC case, the argument stdR can be anything and do not affect the results. That it to say, stdR has to exist but not work in this case.

**Comment by hongyi-zhao** [comment]
> However, when called in noo-SOC case, the argument stdR can be anything and do not affect the results. That it to say, stdR has to exist but not work in this case.

Do you mean that the following usage is sufficient in non-soc situation?

`convTraceToBC[sgno, traceData, P, p, ""]`

**Comment by goodluck1982** [comment]
Yes.

**Comment by goodluck1982** [comment]
> That it to say, stdR has to exist but not work in this case.

Sorry, I forgot I had defined the function overload:
```mathematica
convTraceToBC[sgno_Integer,traceData_,P_,p0_]:=convTraceToBC[sgno,traceData,P,p0,IdentityMatrix[3]]
convTraceToBC[sgno_Integer,traceData_,P_,p0_,stdR_]:=......
```
So, the stdR argument can be omitted. If omitted, it's default value is just the identity matrix.

**Comment by hongyi-zhao** [comment]
So, in non-soc situation, the following is enough:

`convTraceToBC[sgno, traceData, P, p]`

The function called above will be converted to the following form before execution:

`convTraceToBC[sgno,traceData,P,p0,IdentityMatrix[3]]`

**Comment by goodluck1982** [comment]
You are right.
